pkg/pillar/types/zedroutertypes.go

```diff
@@ -3211,7 +3212,7 @@ type AppInstMetaData struct {
 
 // Key : App Instance Metadata unique key
 func (data AppInstMetaData) Key() string {
-	return data.AppInstUUID.String()
+	return data.AppInstUUID.String() + string(data.Type)
```

Comment:
To keep things unified I would put a hyphen in between like it is done in other similar cases, such as here:  https://github.com/lf-edge/eve/blob/8.12/pkg/pillar/types/zedroutertypes.go#L106